d2l/train.py

```diff
@@ -9,19 +9,18 @@
 
 __all__ = ['evaluate_accuracy', 'train_ch3', 'train_ch5']
 
-def evaluate_accuracy(data_iter, net, device=torch.device('cpu')):
+def evaluate_accuracy(data_iter, net,device=torch.device('cpu')):
     """Evaluate accuracy of a model on the given data set."""
-    net.eval()  # Switch to evaluation mode for Dropout, BatchNorm etc layers.
-    acc_sum, n = torch.Tensor([0]).to(device), 0
-    for batch in data_iter:
-        features, labels = batch
-        for X, y in zip(features, labels):
-            # If device is the GPU, copy the data to the GPU.
-            y = y.type(torch.float32)
-            X, y = X.to(device), y.to(device)
-            acc_sum += torch.sum((torch.argmax(net(X), dim=1).type(torch.float32) == y).detach()).float()
-            n += 1
-    return acc_sum.item() / n
+    acc_sum,n = torch.tensor([0],dtype=torch.float32,device=device),0
+    for X,y in data_iter:
+        # If device is the GPU, copy the data to the GPU.
+        X,y = X.to(device),y.to(device)
+        net.eval()
```

Comment:
let the eval mode be at the top(just after the func starts) with the comment. It is more readable there.
```
net.eval()  # Switch to evaluation mode for Dropout, BatchNorm etc layers.
```

Reply:
The following error is resolved.